### DISCUSSION TITLE
Need to install FLAX completely offline with all the dependencies

### DISCUSSION THREAD

**Discussion by abhiroop513** [google/flax#2959] [General]
I am trying to install FLAX in my local directory in a cluster where there is no internet access. No only FLAX but also, the dependencies (like Bazel etc.) would not install offline.
Although JAX has been installed by the administrators, they have disagreed to install FLAX.

So I have to install flax in my local machine and copy the directory to the cluster. Is it possible ? If yes, how should I do that ?
Or else, should I install FLAX offline with all the dependencies ?

**Comment by cgarciae** [comment]
Hey @abhiroop513,

Not an expert on this but managed to do the following:
* `git clone` the flax repo, 
* `cd` into it
* optionally create a virtual env
* Now run `pip wheel . -w wheels`, this creates a `wheels` folders with all the dependencies.
* Copy the `wheels` folder to you other machine.
* Run `pip install --no-index --find-links /path/to/wheels flax` on the other machine to install the dependencies.
